packages/gitbook/src/components/DocumentView/CodeBlock/CodeBlock.tsx

```diff
@@ -1,291 +1,52 @@
-                        withLineNumbers
-                            ? [
-                                  'before:text-dark/5',
-                                  'before:content-[counter(line)]',
-                                  '[counter-increment:line]',
-                                  'dark:before:text-light/4',
-
-                                  line.highlighted
-                                      ? [
-                                            'before:text-dark/6',
-                                            'dark:before:text-light/8',
-                                            'bg-gradient-to-r',
-                                            'from-80%',
-                                            'from-light-3',
-                                            'to-transparent',
-                                            'dark:from-dark-3',
-                                            'dark:to-transparent',
-                                        ]
-                                      : null,
-                              ]
-                            : [],
-                    )}
-                ></span>
-            ) : null}
-
-            <span className={tcls('ml-3', 'block', 'text-sm')}>
-                <CodeHighlightTokens tokens={line.tokens} document={document} context={context} />
-                {isLast ? null : !withLineNumbers && line.tokens.length === 0 && 0 ? (
-                    <span className="ew">{'\u200B'}</span>
-                ) : (
-                    '\n'
-                )}
-            </span>
-        </span>
-    );
-}
-
-function CodeHighlightTokens(props: {
-    tokens: HighlightToken[];
-    document: JSONDocument;
-    context: DocumentContext;
-}) {
-    const { tokens, document, context } = props;
-
-    return (
-        <>
-            {tokens.map((token, index) => (
-                <CodeHighlightToken
+export function CodeBlock(props: BlockProps<DocumentBlockCode>) {
+    const { block, document, style, context, isEstimatedOffscreen } = props;
+    const inlines = getInlines(block);
+    const richInlines: RichInlineIndexed[] = inlines.map((inline, index) => {
+        const body = (() => {
+            const fragment = getNodeFragmentByType(inline.inline, 'annotation-body');
+            if (!fragment) {
+                return null;
+            }
+            return (
+                <Blocks
                     key={index}
-                    token={token}
                     document={document}
+                    ancestorBlocks={[]}
                     context={context}
+                    nodes={fragment.nodes}
+                    style={['space-y-4']}
                 />
-            ))}
-        </>
-    );
-}
+            );
+        })();
 
-function CodeHighlightToken(props: {
-    token: HighlightToken;
-    document: JSONDocument;
-    context: DocumentContext;
-}) {
-    const { token, document, context } = props;
+        return { ...inline, body };
```

Comment:
I usually find it to be quite bad to merge interfaces this way. I'd recommend:

```suggestion
        return { inline, body };
```

It doesn't make the code more complicated and avoid a lot of spread with weird types everywhere.